landlab/utils/source_tracking_algorithm.py

```diff
@@ -0,0 +1,189 @@
+
+"""
+26 Jan 2016
+Authors: Sai Nudurupati & Erkan Istanbulluoglu
+"""
+
+# %%
+# Import required libraries
+import numpy as np
+import copy
+from collections import Counter
+
+# %%
+def convert_arc_flow_directions_to_landlab_node_ids(grid, flow_dir_arc):
+    """
+    Convert Arc flow_directions file to be represented in node ids
+    gives receiver of each node starting at right and going clockwise.
+    
+    Parameters:
+    ----------
+    grid: RasterModelGrid
+        A grid.
+    flow_dir_arc: array_like
+        numpy.array of shape=[grid.number_of_nodes]; flow directions derived
+        from ESRII ArcGIS.
+    
+    Returns:
+    -------
+    receiver_nodes: array_like
+        A numpy.array of shape=[grid.number_of_nodes]. Please note that this
+        array gives the receiver nodes only for the core nodes. For non-core
+        nodes, a zero is used.  
```

Comment:
For this sort of description include the type and shape of the array on the rist line,
```
receiver_nodes : ndarray of int, shape (n_nodes, )
```